src/ManagedNodeAddress.js

```diff
@@ -5,7 +5,7 @@
  * @typedef {import("./address_book/NodeAddress.js").default} NodeAddress
  */
 
-const HOST_AND_PORT = /^(?<address>.*)(:(?<port>\d+))?/;
+const HOST_AND_PORT = /^(.*)(:(\d+))?/;
```

Comment:
Side-note this regex will not catch the port, `.*` will always consume it. (If this is important to you) we should fix it @feri42

Reply:
Looks to me like the original RE never catches to port either.

Reply:
Yes, that is my point. This PR is a pure refactor that removes named groups for compatibility. But while testing I noticed the above & we can fix it separately if needed